virttest/qemu_vm.py

```diff
@@ -1231,13 +1231,14 @@ def add_sga(devices):
                 return " -device sga"
 
         def add_watchdog(devices, device_type=None, action="reset"):
-            watchdog_cmd = ""
-            if devices.has_option("watchdog"):
-                if device_type:
-                    watchdog_cmd += " -watchdog %s" % device_type
-                watchdog_cmd += " -watchdog-action %s" % action
-
-            return watchdog_cmd
+            cmd = " -watchdog-action %s" % action
+            devices.insert(StrDev('watchdog', cmdline=cmd))
+            parent_bus = None
+            if device_type and devices.has_device(device_type):
+                if devices.is_pci_device(device_type):
+                    parent_bus = pci_bus
+                return QDevice(device_type, parent_bus=parent_bus)
```

Comment:
Add `params={'id': watchdog_id}` will be better.
But since we have not scenario like hot unplug watchdog device, so it's up to you.

Reply:
We do not have scenario to add multi watchdog at the same time,  no hotplug scenario as you said.  And we do not define the device id for watchdog in cfg, so i think add a  `utils_misc.generate_random_id()` is not that must. let's keep the current design.